base/steer/FairRootManager.cxx

```diff
@@ -762,38 +754,37 @@ TObject* FairRootManager::ActivateBranch(const char* BrName)
    calls to activate branch is done , and then just forward the pointer.
    <DB>
    **/
-    fNObj++;
-    fObj2[fNObj] = GetMemoryBranch(BrName);
-    if (fObj2[fNObj]) {
-        return fObj2[fNObj];
+    auto& newentry = fObj2.emplace_back(GetMemoryBranch(BrName));
+    if (newentry) {
+        return newentry;
     }
     /**try to find the object decribing the branch in the folder structure in file*/
     LOG(debug) << "Try to find an object " << BrName << " describing the branch in the folder structure in file";
     if (fListFolder) {
         for (Int_t i = 0; i < fListFolder->GetEntriesFast(); i++) {
             TFolder* fold = static_cast<TFolder*>(fListFolder->At(i));
-            fObj2[fNObj] = fold->FindObjectAny(BrName);
-            if (fObj2[fNObj]) {
+            newentry = fold->FindObjectAny(BrName);
+            if (newentry) {
                 LOG(info) << "Object " << BrName << " describing the branch in the folder structure was found";
                 break;
             }
         }
     }
 
-    if (!fObj2[fNObj]) {
+    if (!newentry) {
         /** if we do not find an object corresponding to the branch in the folder structure
          *  then we have no idea about what type of object is this and we cannot set the branch address
          */
         LOG(info) << " Branch: " << BrName << " not found in Tree.";
         // Fatal(" No Branch in the tree", BrName );
-        return 0;
+        return nullptr;
```

Comment:
It looks like `FairRootManager::UpdateBranches` expects all pointers in `fObj2` to be valid. While this looks like it handles a nullptr in `fObj2`.